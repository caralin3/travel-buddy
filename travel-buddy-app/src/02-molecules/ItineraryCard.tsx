import moment from 'moment';
import React from 'react';
import { Row, Col } from 'reactstrap';

export interface ItineraryCardProps {
  date: string;
}

export const ItineraryCard: React.FC<ItineraryCardProps> = ({ children, date }) => (
  <Row className="itinerary-card m-0 mb-5 bg-light">
    <Col xs={12} md={2} lg={1} className="m-0 py-3 bg-dark text-white">
      <span className="itinerary-card__date">{moment(date).format('MM/DD')}</span>
      <br />
      <span className="text-uppercase">{moment(date).format('ddd')}</span>
    </Col>
    <Col xs={12} md={10} lg={11} className="d-flex flex-column justify-content-center px-4 py-3">
      {children}
    </Col>
  </Row>
);
