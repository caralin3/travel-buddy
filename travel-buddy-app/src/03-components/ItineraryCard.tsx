import moment from 'moment';
import React from 'react';
import { Row, Col } from 'reactstrap';
import { Flight, Hotel } from '../api';
import { FlightDetail } from './FlightDetail';
import { HotelDetail } from './HotelDetail';

export interface ItineraryCardProps {
  days: string[];
  flights: Flight[];
  hotels: Hotel[];
}

export const ItineraryCard: React.FC<ItineraryCardProps> = ({ days, flights, hotels }) => {
  const getFlightByDay = (day: string) =>
    flights.filter((flight) => moment(flight.departureDate).format('YYYY-MM-DD') === day);

  const getHotelCheckInByDay = (day: string) =>
    hotels.filter((hotel) => moment(hotel.checkInDate).format('YYYY-MM-DD') === day);

  const getHotelCheckOutByDay = (day: string) =>
    hotels.filter((hotel) => moment(hotel.checkOutDate).format('YYYY-MM-DD') === day);

  return (
    <div>
      {days.map((day) => (
        <Row className="m-0 mb-5 bg-light" key={day}>
          <Col xs={12} md={2} lg={1} className="m-0 py-3 bg-dark text-white">
            <span className="itinerary-card__date">{moment(day).format('MM/DD')}</span>
            <br />
            <span className="text-uppercase">{moment(day).format('ddd')}</span>
          </Col>
          <Col xs={12} md={10} lg={11} className="px-4 py-3">
            {getFlightByDay(day).map((flight) => (
              <div className="itinerary-card__detail">
                <FlightDetail key={flight.id} flight={flight} />
              </div>
            ))}
            {getHotelCheckInByDay(day).map((hotel) => (
              <div className="itinerary-card__detail">
                <HotelDetail key={hotel.checkInDate} hotel={hotel} checkIn />
              </div>
            ))}
            {getHotelCheckOutByDay(day).map((hotel) => (
              <div className="itinerary-card__detail">
                <HotelDetail key={hotel.checkOutDate} hotel={hotel} />
              </div>
            ))}
          </Col>
        </Row>
      ))}
    </div>
  );
};
