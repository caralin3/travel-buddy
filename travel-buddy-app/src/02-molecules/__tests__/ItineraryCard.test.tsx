import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ItineraryCard } from '../ItineraryCard';

describe('ItineraryCard', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ItineraryCard date="2022-03-24" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
