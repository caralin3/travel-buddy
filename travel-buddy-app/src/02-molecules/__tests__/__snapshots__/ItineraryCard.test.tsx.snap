// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryCard should render correctly 1`] = `
<Row
  className="itinerary-card m-0 mb-5 bg-light"
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
      "xxl",
    ]
  }
>
  <Col
    className="m-0 py-3 bg-dark text-white"
    lg={1}
    md={2}
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
    xs={12}
  >
    <span
      className="itinerary-card__date"
    >
      03/24
    </span>
    <br />
    <span
      className="text-uppercase"
    >
      Thu
    </span>
  </Col>
  <Col
    className="d-flex flex-column justify-content-center px-4 py-3"
    lg={11}
    md={10}
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
    xs={12}
  />
</Row>
`;
